src/components/Match/VisionMap.jsx

```diff
@@ -1,136 +1,112 @@
 } from 'utility';
-import Table from 'components/Table';
+import { PlainTable as Table } from 'components/Table';
 import {
   Row,
   Col,
 } from 'react-flexbox-grid';
 // import heroes from 'dotaconstants/json/heroes.json';
 import strings from 'lang';
-import {
-  heroTdColumn,
-} from './matchColumns';
+import styles from './Match.css';
+import { Fixed } from 'utility/components';
+import Measure from 'react-measure';
+
+// with the actual game size, the width parameters is optional
+const style = (width, iconSize, ward) => {
+  const gamePos = gameCoordToUV(ward.x, ward.y);
+  return {
+    position: 'absolute',
+    top: ((width / 127) * gamePos.y) - iconSize/2,
+    left: ((width / 127) * gamePos.x) - iconSize/2,
+  }
+};
 
-const obsWard = (style, stroke, iconSize) => (<svg style={style} width={iconSize} height={iconSize} xmlns="http://www.w3.org/2000/svg">
-  <g>
-    <title>Observer</title>
-    <circle fill="#ffff00" strokeWidth="5" stroke={stroke} r={iconSize * 0.4} cy={iconSize / 2} cx={iconSize / 2} fillOpacity="0.4" />
-  </g>
-  <defs>
-    <filter id="_blur">
-      <feGaussianBlur stdDeviation="0.1" in="SourceGraphic" />
-    </filter>
-  </defs>
-</svg>);
+const WardLogPin = ({ width, iconSize, log }) => {
+  const stroke = log.entered.player_slot < 5 ? styles.green : styles.red;
+  const fill = log.type == "observer" ? styles.yelo : styles.blue;
+  return (
+    <svg style={style(width, iconSize, log.entered)}
+         width={iconSize}
+         height={iconSize}
+         xmlns="http://www.w3.org/2000/svg">
+      <g>
+        <title>Observer</title>
+        <circle fill={fill}
+                strokeWidth="2"
+                stroke={stroke}
+                r={iconSize * 0.4}
+                cy={iconSize / 2}
+                cx={iconSize / 2}
+                fillOpacity="0.4" />
+      </g>
+      <defs>
+        <filter id="_blur">
+          <feGaussianBlur stdDeviation="0.1" in="SourceGraphic" />
+        </filter>
+      </defs>
+    </svg>
+  )
+};
+  
 
-const senWard = (style, stroke, iconSize) => (<svg style={style} width={iconSize} height={iconSize} xmlns="http://www.w3.org/2000/svg">
-  <g>
-    <title>Sentry</title>
-    <circle fill="#0000ff" strokeWidth="5" stroke={stroke} r={iconSize * 0.4} cy={iconSize / 2} cx={iconSize / 2} fillOpacity="0.4" />
-  </g>
-  <defs>
-    <filter id="_blur">
-      <feGaussianBlur stdDeviation="0.1" in="SourceGraphic" />
-    </filter>
-  </defs>
-</svg>);
+// sen = #0000ff;
+
+const FixedTable = Fixed(Table);
 
 // TODO Hero icon on ward circles?
 class VisionMap extends React.Component {
-  componentWillMount() {
-    this.setState({
-      enabledIndex: {},
-    });
+  shouldComponentUpdate(newProps) {
+    if (newProps.wardsLog.length == this.props.wardsLog.length) return false;
+    return true;
   }
-  super() {
-    this.updateMap = this.updateMap.bind(this);
+
+  componentDidMount() {
+    window.addEventListener("resize", () => this.forceUpdate());
```

Comment:
I think @mike-robertson will advise that you use redux-responsive's state.browser.width for this